### ISSUE TITLE
select_ipc fails with TApplicationException.MISSING_RESULT in presence of bool column

### ISSUE THREAD

**Issue by randyzwitch** [omnisci/pymapd#211]
As part of #203, tested a table with the following structure:

```bash
tinyint_ TINYINT
smallint_ SMALLINT
int_ INT
bigint_ BIGINT
float_ FLOAT
double_ DOUBLE
bool_ BOOL
```

Tests without the bool column run fine; adding the bool column and using `select_ipc` throws a `TApplicationException(TApplicationException.MISSING_RESULT, "sql_execute failed: unknown result") ` error. So the failure is at reading the Arrow buffer at the Thrift stage, not a pymapd parsing issue.

There are no nulls in any column of the table.

pymapd current master (post 0.10)
pyarrow 0.12.1
python 3.6.7

```python
debugfile('/home/rzwitch/Desktop/untitled0.py', wdir='/home/rzwitch/Desktop')
> /home/rzwitch/Desktop/untitled0.py(7)<module>()
      5 
      6 @author: rzwitch
----> 7 """
      8 
      9 


ipdb> > /home/rzwitch/Desktop/untitled0.py(17)<module>()
     15 
     16 
3--> 17 df = con.select_ipc("select * from test_data_no_nulls")
     18 
     19 


ipdb> --Call--
> /home/rzwitch/pymapd/pymapd/connection.py(286)select_ipc()
    284         return df
    285 
--> 286     def select_ipc(self, operation, parameters=None, first_n=-1,
    287                    release_memory=True):
    288         """Execute a ``SELECT`` operation using CPU shared memory


ipdb> > /home/rzwitch/pymapd/pymapd/connection.py(310)select_ipc()
    308         """
    309 
--> 310         if parameters is not None:
    311             operation = str(_bind_parameters(operation, parameters))
    312 


ipdb> > /home/rzwitch/pymapd/pymapd/connection.py(313)select_ipc()
    311             operation = str(_bind_parameters(operation, parameters))
    312 
--> 313         tdf = self._client.sql_execute_df(
    314             self._session, operation.strip(), device_type=0, device_id=0,
    315             first_n=first_n


ipdb> > /home/rzwitch/pymapd/pymapd/connection.py(314)select_ipc()
    312 
    313         tdf = self._client.sql_execute_df(
--> 314             self._session, operation.strip(), device_type=0, device_id=0,
    315             first_n=first_n
    316         )


ipdb> > /home/rzwitch/pymapd/pymapd/connection.py(315)select_ipc()
    313         tdf = self._client.sql_execute_df(
    314             self._session, operation.strip(), device_type=0, device_id=0,
--> 315             first_n=first_n
    316         )
    317         self._tdf = tdf


ipdb> --Call--
> /home/rzwitch/pymapd/mapd/MapD.py(1550)sql_execute_df()
   1548         raise TApplicationException(TApplicationException.MISSING_RESULT, "sql_execute failed: unknown result")
   1549 
-> 1550     def sql_execute_df(self, session, query, device_type, device_id, first_n):
   1551         """
   1552         Parameters:


ipdb> 
ipdb> > /home/rzwitch/pymapd/mapd/MapD.py(1559)sql_execute_df()
   1557          - first_n
   1558         """
-> 1559         self.send_sql_execute_df(session, query, device_type, device_id, first_n)
   1560         return self.recv_sql_execute_df()
   1561 


ipdb> > /home/rzwitch/pymapd/mapd/MapD.py(1560)sql_execute_df()
   1558         """
   1559         self.send_sql_execute_df(session, query, device_type, device_id, first_n)
-> 1560         return self.recv_sql_execute_df()
   1561 
   1562     def send_sql_execute_df(self, session, query, device_type, device_id, first_n):


ipdb> --Call--
> /home/rzwitch/pymapd/mapd/MapD.py(1574)recv_sql_execute_df()
   1572         self._oprot.trans.flush()
   1573 
-> 1574     def recv_sql_execute_df(self):
   1575         iprot = self._iprot
   1576         (fname, mtype, rseqid) = iprot.readMessageBegin()


ipdb> > /home/rzwitch/pymapd/mapd/MapD.py(1575)recv_sql_execute_df()
   1573 
   1574     def recv_sql_execute_df(self):
-> 1575         iprot = self._iprot
   1576         (fname, mtype, rseqid) = iprot.readMessageBegin()
   1577         if mtype == TMessageType.EXCEPTION:


ipdb> > /home/rzwitch/pymapd/mapd/MapD.py(1576)recv_sql_execute_df()
   1574     def recv_sql_execute_df(self):
   1575         iprot = self._iprot
-> 1576         (fname, mtype, rseqid) = iprot.readMessageBegin()
   1577         if mtype == TMessageType.EXCEPTION:
   1578             x = TApplicationException()


ipdb> thrift.transport.TTransport.TTransportException: TSocket read 0 bytes
> /home/rzwitch/pymapd/mapd/MapD.py(1576)recv_sql_execute_df()
   1574     def recv_sql_execute_df(self):
   1575         iprot = self._iprot
-> 1576         (fname, mtype, rseqid) = iprot.readMessageBegin()
   1577         if mtype == TMessageType.EXCEPTION:
   1578             x = TApplicationException()


ipdb> --Return--
None
> /home/rzwitch/pymapd/mapd/MapD.py(1576)recv_sql_execute_df()
   1574     def recv_sql_execute_df(self):
   1575         iprot = self._iprot
-> 1576         (fname, mtype, rseqid) = iprot.readMessageBegin()
   1577         if mtype == TMessageType.EXCEPTION:
   1578             x = TApplicationException()

thrift.transport.TTransport.TTransportException: TSocket read 0 bytes
> /home/rzwitch/pymapd/mapd/MapD.py(1560)sql_execute_df()
   1558         """
   1559         self.send_sql_execute_df(session, query, device_type, device_id, first_n)
-> 1560         return self.recv_sql_execute_df()
   1561 
   1562     def send_sql_execute_df(self, session, query, device_type, device_id, first_n):


ipdb> 
ipdb> 
```

**Comment by randyzwitch** [comment]
Patched in https://github.com/omnisci/mapd-core-internal/pull/3281, will leave open until patch is in stable release and test written.

**Comment by andrewseidl** [comment]
Fix is in public master at https://github.com/omnisci/mapd-core/commit/10c47fb4947d601624e65c1f028e6e051e5efbd0

Might not make it into v4.6.0, but should be in v4.6.1 and will be in v4.7.0.

**Comment by randyzwitch** [comment]
4.6.1 released, proven fixed by #225
